### PR TITLE
Run Continuous Integration tests on Github Actions

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Newer attempt of https://github.com/weiji14/pygmt/pull/35.

Fixes #

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

**Comment by seisman** [comment]
The GitHub Actions look promising. It seems the macOS agent doesn't correctly cache the `~/.gmt` directory (i.e., the cache size is only 22 B). Perhaps you could try to update the cache key to re-cache the directory?

**Comment by weiji14** [comment]
> The GitHub Actions look promising.

Yeah, it's meant to consolidate all our Linux/macOS/Windows unit tests into one file using a matrix build specification (instead of having .travis.yml and .azure-pipelines.yml). Was going to put gmt-master into it somehow too at some point. It's been a lot of trial and error at this point getting it to work but should be getting there!

> It seems the macOS agent doesn't correctly cache the `~/.gmt` directory (i.e., the cache size is only 22 B). Perhaps you could try to update the cache key to re-cache the directory?

Good spotting, I'll try that. Not sure if it's because the Linux build runs faster than macOS and 'caches' things first.

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      - name: List installed packages
+        shell: bash -l {0}
+        run: conda list
+
+      # Cache the ${HOME}/.gmt directory, for docs and testing
+      - name: Cache GMT directory
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: $(HOME)/.gmt
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:50]: GitHub Actions may not support `$(HOME)`. Try `~/.gmt` instead. See the [pip cache example](https://github.com/actions/cache/blob/master/examples.md#python---pip)

Comment (weiji14) [reply .github/workflows/ci_tests.yaml:50]: Ok, won't hurt to try.

Comment (seisman) [reply .github/workflows/ci_tests.yaml]: Now macOS has the correct cache (~6MB), but Linux fails to extract the cache tarballs.

Comment (weiji14) [reply .github/workflows/ci_tests.yaml]: Yeah. Not sure why the Windows run is saying it can't find the cache.

Comment (weiji14) [reply .github/workflows/ci_tests.yaml]: I'll try using `${{ github.workspace }}/.gmt` instead.

Comment (seisman) [reply .github/workflows/ci_tests.yaml]: Still no luck. I noticed that Linux and Windows agents actually can download some files but timeout for others. For example:
```
Run gmt which -Gu @earth_relief_10m @earth_relief_60m @ridge.txt @Table_5_11.txt @tut_bathy.nc @tut_quakes.ngdc @tut_ship.xyz
earth_relief_10m: Download file from the GMT data server [data set size is 2.8M].
earth_relief_10m: Earth Relief at 10x10 arc minutes obtained by Gaussian Cartesian filtering (18 km fullwidth) of SRTM15+V2 [Tozer et al., 2019].

gmtwhich [ERROR]: Libcurl Error: Timeout was reached
gmtwhich [ERROR]: You can turn remote file download off by setting GMT_DATA_SERVER_LIMIT = 0.
gmtwhich [ERROR]: File earth_relief_10m.grd not found!
earth_relief_60m: Download file from the GMT data server [data set size is 106K].
earth_relief_60m: Earth Relief at 60x60 arc minutes obtained by Gaussian Cartesian filtering (111 km fullwidth) of SRTM15+V2 [Tozer et al., 2019].

C:/Users/runneradmin/.gmt/server/earth_relief_60m.grd
C:/Users/runneradmin/.gmt/cache/ridge.txt
C:/Users/runneradmin/.gmt/cache/Table_5_11.txt
C:/Users/runneradmin/.gmt/cache/tut_bathy.nc
C:/Users/runneradmin/.gmt/cache/tut_quakes.ngdc
C:/Users/runneradmin/.gmt/cache/tut_ship.xyz
```
I think a workaround would be running the "gmt which" command multiple times until all files are downloaded.

File: .github/workflows/ci_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+jobs:
+  test:
+    name: ${{ matrix.os }} - Python ${{ matrix.python_version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 3
+      fail-fast: false
+      matrix:
+        python_version: [3.7]
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:20]: `matrix.python-version` is used in the setup-miniconda action, not `matrix.python_version`.
```suggestion
        python-version: [3.7]
```

Comment (weiji14) [reply .github/workflows/ci_tests.yaml]: Good catch, I'll need to change this for one of my other repos as well!

File: .github/workflows/ci_tests.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    name: ${{ matrix.os }} - Python ${{ matrix.python_version }}
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:14]: Another typo:
```suggestion
    name: ${{ matrix.os }} - Python ${{ matrix.python-version }}
```

Comment (weiji14) [reply .github/workflows/ci_tests.yaml:14]: I see you've been working on it on a branch in upstream `pygmt`. Should I move this Pull Request there so we can both work on it?

Comment (seisman) [reply .github/workflows/ci_tests.yaml:14]: Yes, that would be better.

**Comment by weiji14** [comment]
Closing in favour of upstream PR at https://github.com/GenericMappingTools/pygmt/pull/475.

**Pull request closed without merging**